prow/tide/tide.go

```diff
@@ -944,7 +954,7 @@ func (c *Controller) presubmitsByPull(sp subpool) (map[int]sets.String, error) {
 	}()
 
 	for _, ps := range c.ca.Config().Presubmits[sp.org+"/"+sp.repo] {
-		if ps.SkipReport || !ps.RunsAgainstBranch(sp.branch) {
+		if !ps.ContextRequired() || !ps.RunsAgainstBranch(sp.branch) {
```

Comment:
Why are we changing this functionality? If we want to do this, let's do it in a separate PR?

Reply:
This is the goal of this PR

Reply:
The reason is that SkipReport = false does not mean the check is required. In our use case you might want to push a status on a PR for a test that is not required, this is the reason I added Optional a while back. Optional is false by default so it will not impact anything, but then if you define a prowjob with Optional set to true, you defintely want to skip it to merge a PR. So this is the reason I added this.

Reply:
I understand they are not identical features, what I am saying is today we use `SkipReport` to mean "tide ignore" because whether or not it's required, if it's not reported, `tide` ignores it. As long as `SkipReport=true` continues to mean that `tide` ignores the job, but now in addition, `SkipReport=false Optional=true` works, that is OK.

Reply:
ok.